consensus/safety-rules/src/persistent_safety_storage.rs

```diff
@@ -109,14 +113,17 @@ impl PersistentSafetyStorage {
     }
 
     pub fn safety_data(&self) -> Result<SafetyData, Error> {
-        Ok(self.internal_store.get(SAFETY_DATA).map(|v| v.value)?)
+        Ok(self.cached_safety_data.clone())
```

Comment:
This doesn't seem to ever read the safety_data from Vault. I think the cache should be `Option<SafetyData>`, and this function must read from Vault if it is `none`.

Reply:
Aah, thanks for pointing this out @mgorven! I wasn't aware we had a use case where we could load the SafetyData from storage upon startup, but in hindsight it's clear we need to support for that failure-recovery cases 🤦. I've updated the code so that we only use the cache if the flag is enabled and the cache is set.